File: 0052-n-queens-ii/0052-n-queens-ii.java
public class Solution {
    private int size;
    private int count;

    public int totalNQueens(int n) {
        count = 0;
        size = (1 << n) - 1;
        solve(0, 0, 0);
        return count;
    }

    private void solve(int queens, int ld, int rd) {
        if (queens == size) {
            count++;
            return;
        }
        int pos = size & ~(queens | ld | rd);
        while (pos > 0) {
            int bit = pos & -pos;
            pos -= bit;
            solve(queens | bit, (ld | bit) << 1, (rd | bit) >> 1);
        }
    }
}

File: 0052-n-queens-ii/0052-n-queens-ii.js
/**
 * @param {number} n
 * @return {number}
 */
var totalNQueens = function(n) {
    let count = 0;

    const DFS = (queens, xyDiff, xySum) => {
        let p = queens.length;
        if (p == n) {
            count++;
            return;
        }
        for (let q = 0; q < n; q++) {
            if (!queens.includes(q) && !xyDiff.includes(p-q) && !xySum.includes(p+q)) {
                DFS(queens.concat(q), xyDiff.concat(p-q), xySum.concat(p+q));
            }
        }
    }

    DFS([], [], []);
    return count;
};
